crates/common/fmt/src/ui.rs

```diff
@@ -306,10 +306,143 @@ impl UIfmt for OtherFields {
     }
 }
 
+impl UIfmt for AccessListItem {
+    fn pretty(&self) -> String {
+        self.storage_keys.pretty()
+    }
+}
+
 impl UIfmt for Transaction {
     fn pretty(&self) -> String {
-        format!(
-            "
+        match self.transaction_type {
```

Comment:
this is a bit verbose, but I'm okay with this because we want to eventually embed the actual tx object in the rpc response and then we need to do this anyway